components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OASParserUtil.java

```diff
@@ -426,48 +427,66 @@ public static JsonNode generateOASConfigForEndpoints(API api, boolean isProducti
             // no need to populate if it is prototype API
             return null;
         }
-        // Validation for only one endpoint type is provided.
-        if (isProduction && !endpointConfig.has(APIConstants.ENDPOINT_PRODUCTION_ENDPOINTS)) {
-            return null;
-        }
-        if (!isProduction && !endpointConfig.has(APIConstants.ENDPOINT_SANDBOX_ENDPOINTS)) {
-            return null;
-        }
-        ObjectNode endpointResult = objectMapper.createObjectNode();
+        ObjectNode endpointResult;
         String type = endpointConfig.getString(APIConstants.API_ENDPOINT_CONFIG_PROTOCOL_TYPE);
         if (APIConstants.ENDPOINT_TYPE_DEFAULT.equalsIgnoreCase(type)) {
+            endpointResult = objectMapper.createObjectNode();
             endpointResult.put(APIConstants.X_WSO2_ENDPOINT_TYPE, APIConstants.ENDPOINT_TYPE_DEFAULT);
-            return endpointResult;
         } else if (APIConstants.ENDPOINT_TYPE_FAILOVER.equalsIgnoreCase(type)) {
-            populateFailoverConfig(endpointResult, endpointConfig, isProduction);
+            endpointResult = populateFailoverConfig(endpointConfig, isProduction);
         } else if (APIConstants.ENDPOINT_TYPE_LOADBALANCE.equalsIgnoreCase(type)) {
-            populateLoadBalanceConfig(endpointResult, endpointConfig, isProduction);
+            endpointResult = populateLoadBalanceConfig(endpointConfig, isProduction);
         } else if (APIConstants.ENDPOINT_TYPE_HTTP.equalsIgnoreCase(type)) {
-            setPrimaryConfig(endpointResult, endpointConfig, isProduction, APIConstants.ENDPOINT_TYPE_HTTP);
+            endpointResult = setPrimaryConfig(endpointConfig, isProduction, APIConstants.ENDPOINT_TYPE_HTTP);
         } else if (APIConstants.ENDPOINT_TYPE_ADDRESS.equalsIgnoreCase(type)) {
-            setPrimaryConfig(endpointResult, endpointConfig, isProduction, APIConstants.ENDPOINT_TYPE_ADDRESS);
+            endpointResult = setPrimaryConfig(endpointConfig, isProduction, APIConstants.ENDPOINT_TYPE_ADDRESS);
         } else {
             return null;
         }
+        if(endpointResult != null) {
```

Comment:
need to format.

Reply:
Fixed in https://github.com/wso2/carbon-apimgt/pull/7500